web/src/message_lists.ts

```diff
@@ -130,7 +131,38 @@ export function save_pre_narrow_offset_for_reload(): void {
     }
 }
 
+export function calculate_timestamp_widths(): void {
+    const $temp_time_div = $("<div>");
+    $temp_time_div.attr("id", "calculated-timestamp-widths");
+    // Size the div to the width of the largest timestamp
+    $temp_time_div.css({width: "max-content", visibility: "hidden"});
+    // We should get a reasonable max-width by looking only at
+    // the first and last minutes of AM and PM
+    const candidate_times = ["00:01", "11:59", "13:01", "23:59"];
```

Comment:
The first and last minutes of AM and PM are

```suggestion
    const candidate_times = ["00:00", "11:59", "12:00", "23:59"];
```